### PR TITLE
Feat: new syntax highlighting tutorial

### PR DESCRIPTION
**Pull request by bholmesdev**
## What's changed?
- Make Shiki the default under our "Syntax Highlighting" advice
- Update config examples for this new default
- Explain how custom themes work: create a JSON token list, try CSS variables

### CODE REVIEW

**Comment by sarah11918** [comment]
I'm taking a closer look at this, but after an initial look over, my first Q for @FredKSchott is: **is this more detail than you want in the plain-old "Markdown Guide", and is some of this more of reference/config material?**

@bholmesdev - we've tended to keep the content on these guide pages a little more introductory, closer to "how to make sure you HAVE syntax highlighting" vs how to customize the heck out of it. So, I'll go through this while also considering what might be considered more advanced usage for pulling out and adding to the "Reference" section of the docs.  :)

**Comment by sarah11918** [comment]
Admittedly, I haven't given much thought yet as to WHERE exactly it would go... But we do eventually want more How-To walk through type guide/tutorial material, so I think eventually absolutely, it will want to be its own, cool thing.

So, I don't want to lose the content, for sure! It just might live out of a suitcase temporarily.  :)

**Comment by bholmesdev** [comment]
@sarah11918 Ah okay! Was _hoping_ to get it added sooner rather than later to help Prism users migrate their custom stylesheets. Could you see this living under the "Reference" tab so we can link it from the 0.25 upgrade guide?

File: src/pages/en/guides/markdown-content.md [hunk 1]
```diff
@@ -355,11 +355,18 @@ export default {
 
 ### Syntax Highlighting
 
-Astro comes with built-in support for [Prism](https://prismjs.com/) and [Shiki](https://shiki.matsu.io/). 
+Astro comes with built-in support for [Shiki](https://shiki.matsu.io/) and [Prism](https://prismjs.com/). 
 
-By default, Prism is enabled but no Prism stylesheet is included. (Choose from the available [Prism Themes](https://github.com/PrismJS/prism-themes) and see the [list of languages supported by Prism](https://prismjs.com/#supported-languages) for options and usage.)
+Shiki is enabled by default, preconfigured with the `github-dark` theme. This should provide instant syntax highlighting for all code fences (\`\`\`) used in a markdown (`.md`) file or the [built-in `<Markdown />` component](#markdown-component). What's more, the compiled output will be limited to inline `style`s without any extraneous CSS classes, stylesheets, or client-side JS.
```
Comment (sarah11918) [inline src/pages/en/guides/markdown-content.md:361]: "should" makes me nervous! 😅 

Is it correct to say that whichever you have enabled, Prism or Shiki, controls the syntax highlighting in both the plain markdown and the Markdown component?

I would be more likely to start with WHAT is controlled, then say HOW Shiki does it (specifying it's the default). Maybe something like:

Astro comes with built-in support for [Shiki](https://shiki.matsu.io/) and [Prism](https://prismjs.com/) for instant syntax highlighting for all code fences (\`\`\`) used in a markdown (`.md`) file and in the [built-in `<Markdown />` component](#markdown-component). What's more, the compiled output will be limited to inline `style`s without any extraneous CSS classes, stylesheets, or client-side JS.

Shiki is enabled by default, preconfigured with the `github-dark` theme.

File: src/pages/en/guides/markdown-content.md [hunk 2]
```diff
@@ -368,22 +375,120 @@ export default {
...
+          wrap: true,
         },
       },
     ],
   },
 };
 ```
 
-You can read more about custom Shiki [themes](https://github.com/shikijs/shiki/blob/main/docs/themes.md#loading-theme) and [languages](https://github.com/shikijs/shiki/blob/main/docs/languages.md#supporting-your-own-languages-with-shiki).
+Also consider these resources for extended reading:
```
Comment (sarah11918) [inline src/pages/en/guides/markdown-content.md:397]: We need an explicit warning note SOMEWHERE that if you choose Prism, you also need to add a Prism theme. (Prism will add the classes for you, but requires one of their own themes to define the classes.)

File: src/pages/en/guides/markdown-content.md [hunk 3]
```diff
@@ -368,22 +375,120 @@ export default {
...
+We suggest [using an existing theme](https://github.com/shikijs/shiki/blob/main/scripts/themeSources.ts#L40-L68) as a base before getting started. This way, you can find-and-replace with your desired styling before diving _too_ deeply into all available options (spoiler: there's quite a few!)
+
+You can also [check the custom theme for this very website](https://github.com/withastro/docs/blob/main/syntax-highlighting-theme.ts) (docs.astro.build) to see how we set everything up.
+
+##### Option 2: use the `css-variables` theme
+
+If maintaining a large JSON object feels daunting, you can always try Shiki's `css-variables` theme. Enable this using the `theme` configuration option:
+
+```js
+// astro.config.mjs
```
Comment (sarah11918) [inline src/pages/en/guides/markdown-content.md:459]: Just pointing out that if we decide to go through and change `.mjs` to `.js`, then this gets updated. (More reason to KEEP using `.mjs` for now, I think!)

**Comment by sarah11918** [comment]
Couple of comments above, and I'll leave it up to @FredKSchott as to exactly where he'd like the final section re: custom theme to go... but totally agree, it should go SOMEWHERE! (Just not convinced it goes exactly on this page.)  :)

Thanks for this, @bholmesdev - I know I'll be needing this when I eventually switch over myself! I'm sure there are others like me still on Prism by default (maybe the majority!), so I want to make sure that this page is for the moment slightly Markdown-first/Shiki-second, while we still have several users who will never have swapped their default configuration in the first place. I'm sure there are a TON of users who don't even know that we got the option to switch TO Shiki, let alone are expecting this used as the default. So, there will still be some Prism to support in the transition period.

File: src/pages/en/guides/markdown-content.md (hunk 1 above)
Comment (bholmesdev) [reply src/pages/en/guides/markdown-content.md:361]: Erm, that rewrite is somewhat correct. The "compiled output..." line is Shiki-specific. We'll likely need a separate explainer for Prism. Pushing up a rewrite momentarily.

File: src/pages/en/guides/markdown-content.md (hunk 2 above)
Comment (bholmesdev) [reply src/pages/en/guides/markdown-content.md:397]: Will be adding this into the refactor I mentioned above ☝️

File: src/pages/en/guides/markdown-content.md (hunk 1 above)
Comment (sarah11918) [reply src/pages/en/guides/markdown-content.md:361]: Gotcha! Make it accurate, for sure!

I just would prefer that we say as much that is generic/common first, THEN talk about Shiki. Do whatever is correct!

Comment (bholmesdev) [reply src/pages/en/guides/markdown-content.md:361]: Well hate to say it, but... almost nothing is in common between these 2 tools 😆 Only similarity is "they do syntax highlighting" as the section heading implies. It'll sadly take 2 different explainers.

Comment (sarah11918) [reply src/pages/en/guides/markdown-content.md:361]: Ha! But, to clarify, whichever one you set **does** do **both** the fenced code in Markdown, and the Markdown component, right?

Even if the commonalities are short, I would be tempted to at least start the section something like: 

Astro comes with built-in support for using either [Shiki](https://shiki.matsu.io/) or [Prism](https://prismjs.com/) for instant syntax highlighting for all code fences (```) used in a markdown (.md) file and in the [built-in <Markdown /> component](https://github.com/withastro/docs/pull/237#markdown-component).

THEN, by all means, go wild with Shiki, and make Prism an afterthought. I think if we're mentioning that there are two options, (and, both can actually be used in the same project, given that you can use the Prism and Code components) then it should be clear upfront exactly what setting the config actually affects. (I'm assuming, both those two types of code mentioned above?)

**Comment by sarah11918** [comment]
Looks great!

**Pull request merged**